doc/source/contributor/package-updates.rst

```diff
@@ -0,0 +1,202 @@
+=============================
+Package and Container Updates
+=============================
+
+Preparations
+============
+
+1. Before building images, you should check for any outstanding PRs into the “base branch“. Below are the links for the Wallaby branches.
+
+ kayobe-config: https://github.com/stackhpc/stackhpc-kayobe-config/pulls?q=is%3Apr+is%3Aopen+base%3Astackhpc%2Fwallaby
+
+ kolla: https://github.com/stackhpc/kolla/pulls?q=is%3Apr+is%3Aopen+base%3Astackhpc%2Fwallaby
+
+ kolla-ansible: https://github.com/stackhpc/kolla-ansible/pulls?q=is%3Apr+is%3Aopen+base%3Astackhpc%2Fwallaby
+
+ You should also check any referenced source trees in etc/kayobe/kolla.yml.
+
+ e.g: https://github.com/stackhpc/stackhpc-kayobe-config/blob/stackhpc/wallaby/etc/kayobe/kolla.yml#L112-L158
+
+2. Follow the workflows documented `here <https://stackhpc.github.io/stackhpc-release-train/usage/content-howto/#update-package-repositories>`_. Sync the package repositories. Then, for each release: update the Kayobe package repository versions, build and push Kolla container images, open a draft PR with the updated container image tags. The rest of this document describes the stage "Test".
+
+Testing
+=======
+
+The following steps describe the process to test the new package and container repositories. See the subsections beneath for further explanations.
+
+1. Build two multinode environments for OVS and OVN, both on the "base branch".
+
+2. Run tests on current package versions as a baseline.
+
+3. Upgrade host packages.
+
+4. Upgrade containers.
+
+5. Run tests again with the new packages.
+
+6. Request reviews for your proposed PR to bring in the new packages.
+
+7. Promote these packages before merging the PR, but after the CI checks have passed.
+
+8. Upgrade OpenStack to the next release.
+
+9.  Repeat steps 2 and 4-7. (Step 3 is skipped as the host packages will be shared across these releases.)
+
+10. Repeat 8 and 9 for any further releases.
+
+Creating the multinode environments
+-----------------------------------
+
+There is a comprehensive guide to setting up a multinode environment with Terraform, found here: https://github.com/stackhpc/terraform-kayobe-multinode. There are some things to note:
+
+* OVN is enabled by default, you should override it under ``etc/kayobe/environments/ci-multinode/kolla.yml kolla_enable_ovn: false`` for the OVS multinode environment.
+
+* Remember to set different vxlan_vnis for each.
+
+* Before running deploy-openstack.sh, run distro sync on each host to ensure you are using the same snapshots as in the release train.
```

Comment:
```suggestion
* Before running deploy-openstack.sh, run `dnf distro-sync` on each host to ensure you are using the same snapshots as in the release train. You can do this using the following commands:

  * `kayobe seed host command run -b --command "dnf distro-sync"`
  * `kayobe overcloud host command run -b --command "dnf distro-sync"`
```

Reply:
Actually, if we do this before running deploy-openstack.sh, will the release train package repos be configured on the hosts?

Reply:
No they won't have been, as `kayobe seed/overcloud host configure` both haven't been run yet. I think it should be ok to run these after the deploy-openstack script, I'll just need to clarify it should be done before any testing starts.

Reply:
I agree it would be good to add it to deploy-openstack.sh, removing manual steps is preferable :) . I'll set up a PR for that, and add instruction on how to check the kernel version.

Reply:
Should we aim to have this merged? https://github.com/stackhpc/stackhpc-kayobe-config/pull/377
It's useful for making the release notes and would help here to discover which kernel version is the latest.

Reply:
Will add an alternative for now. After running the distrosync, we can run `dnf list installed kernel` and `uname -a` to check if the running version is the same as the latest installed version.